src/Generation/ResourcesGenerator.php

```diff
@@ -66,4 +69,71 @@ public static function generateDescriptorConfig(ServiceDetails $serviceDetails):
 
         return "<?php\n\n{$return->toCode()};";
     }
+
+    public static function generateClientConfig(ServiceDetails $serviceDetails, GrpcServiceConfig $grpcServiceConfig): string
+    {
+        $inc = fn($i) => $i + 1;
+        $durationToMillis = fn($d) => (int)($d->getSeconds() * 1000 + $d->getNanos() / 1e6);
+
+        $retryCodes = $grpcServiceConfig->retryPolicies
+            ->map(fn($x, $i) => ["retry_policy_{$inc($i)}_codes", is_null($x) ? null :
+                Vector::new($x->getRetryableStatusCodes())->map(fn($x) => Code::name($x))->toArray()])
+            ->filter(fn($x) => !is_null($x[1]))
+            ->append(['no_retry_codes', []])
+            ->toArray(fn($x) => $x[0], fn($x) => $x[1]);
+
+            $retryParams = Vector::zip($grpcServiceConfig->retryPolicies, $grpcServiceConfig->timeouts, fn($r, $t, $i) =>
```

Comment:
This section looks like it's indented too far.

Reply:
Thanks. Fixed.